### PR TITLE
Subprojects to Finance-Publishing

### PR DESCRIPTION
**Pull request by Charlie-Henry**
Config for https://github.com/cityofaustin/atd-data-tech/issues/12376

Will end up being finance DB -> S3 -> Knack pipeline.

### CODE REVIEW

File: upload_to_s3.py [hunk 1]
```diff
@@ -29,11 +29,104 @@
 # don't care about but also because any datetime fields would require extra handling in
 # order to JSON-serialize them
 QUERIES = {
-    "task_orders": "select atd_tk.TASK_ORDER_DEPT, atd_tk.TASK_ORDER_ID, atd_tk.TASK_ORDER_DESC, atd_tk.TASK_ORDER_STATUS, atd_tk.TASK_ORDER_TYPE, atd_tk.TK_CURR_AMOUNT, atd_tk.CHARGED_AMOUNT, atd_tk.TASK_ORDER_BAL, atd_tk.TASK_ORDER_ESTIMATOR, buyer_tk.BYR_FDU FROM DEPT_2400_TK_VW atd_tk LEFT JOIN( SELECT DISTINCT TASK_ORD_CD, BYR_FDU FROM REL_BUYER_SELLER_FDU_TK) buyer_tk ON atd_tk.TASK_ORDER_ID = buyer_tk.TASK_ORD_CD where TASK_ORDER_STATUS is not null",
-    "units": "select DEPT_UNIT_ID, DEPT_ID, DEPT, UNIT, UNIT_LONG_NAME, UNIT_SHORT_NAME, DEPT_UNIT_STATUS from lu_dept_units WHERE DEPT in(2400,6207,2507)",
-    "objects": "select OBJ_ID, OBJ_CLASS_ID, OBJ_CATEGORY_ID, OBJ_TYPE_ID, OBJ_GROUP_ID, OBJ_CODE, OBJ_LONG_NAME, OBJ_SHORT_NAME, OBJ_DESC, OBJ_REIMB_ELIG_STATUS, OBJ_STATUS, ACT_FL from lu_obj_cd",
-    "master_agreements": "select DOC_CD, DOC_DEPT_CD, DOC_ID, DOC_DSCR, DOC_PHASE_CD, VEND_CUST_CD, LGL_NM from DEPT_2400_MA_VW",
-    "fdus": "select * from ATD_SUBPROJECT_FDU_VW",
```
Comment (Charlie-Henry) [inline upload_to_s3.py:36]: Just reformatting these

File: upload_to_s3.py [hunk 2]
```diff
@@ -29,11 +29,104 @@
...
+        SP_NUMBER_TXT,
+        SP_NAME,
+        SP_DESCRIPTION,
+        SP_DETAILED_SCOPE,
+        SUB_PROJECT_MANAGER,
+        SUB_PROJECT_MANAGING_DEPT,
+        SP_STATUS
+    FROM
+        MSTR_IA_DEV.DEPT_2400_SUBPRJ_VW
+	""",
```
Comment (Charlie-Henry) [inline upload_to_s3.py]: new query

File: upload_to_s3.py [hunk 3]
```diff
@@ -88,8 +186,14 @@ def main():
...
-    session = boto3.session.Session(aws_access_key_id=AWS_ACCESS_KEY_ID,aws_secret_access_key=AWS_SECRET_ACCESS_KEY)
-    client = session.client("s3",aws_access_key_id=AWS_ACCESS_KEY_ID,aws_secret_access_key=AWS_SECRET_ACCESS_KEY)
+    session = boto3.session.Session(
+        aws_access_key_id=AWS_ACCESS_KEY_ID, aws_secret_access_key=AWS_SECRET_ACCESS_KEY
+    )
+    client = session.client(
+        "s3",
+        aws_access_key_id=AWS_ACCESS_KEY_ID,
+        aws_secret_access_key=AWS_SECRET_ACCESS_KEY,
+    )
```
Comment (Charlie-Henry) [inline upload_to_s3.py:94]: black styling

**Review by chiaberry — APPROVED** [review]
thanks for reformatting the query, makes it easier to read for sure. 

didn't run the code, but 👍🏼

File: upload_to_s3.py (hunk 2 above)
Comment (johnclary) [reply upload_to_s3.py]: thanks for linting these queries: it might be time to move them to a separate file?

File: upload_to_s3.py [hunk 4]
```diff
@@ -99,4 +203,3 @@ def main():
 if __name__ == "__main__":
     logging.basicConfig(stream=sys.stdout, level=logging.INFO)
     main()
-
```
Comment (johnclary) [inline upload_to_s3.py]: would you add this newline back? or have Black do it 😇

**Review by johnclary — APPROVED** [review]
Taking your word for it that this is works: it looks very excellent to me!

@Charlie-Henry it's fine if this is outta scope, but FWIW it would be helpful for Moped if this data also lived in Socrata :)

**Comment by Charlie-Henry** [comment]
@johnclary no problem to make a Socrata dataset. Do you need the name fields in Socrata? `SUB_PROJECT_LAST_UPDATE_BY`,  `SUB_PROJECT_MANAGER` Are the only fields I think we shouldn't expose
Example [here](https://github.com/cityofaustin/atd-data-tech/issues/12376#issuecomment-1649922167)

**Comment by johnclary** [comment]
@Charlie-Henry thanks for flagging those fields—good call on excluding them. i agree we shouldn't push those into the open data portal.

**Pull request merged**